AdyenComponents/Boleto/BoletoComponent.swift

```diff
@@ -0,0 +1,233 @@
+                postfix: localizedString(key, localizationParameters)
+            )
+        ).withPadding(padding: .init(top: 8, left: 0, bottom: 0, right: 0))
+    }
+    
+    /// :nodoc:
+    private lazy var formComponent: FormComponent = {
+        let component = FormComponent(
+            paymentMethod: paymentMethod,
+            configuration: AbstractPersonalInformationComponent.Configuration(fields: getFormFields()),
+            onCreatePaymentDetails: { [weak self] in self?.createPaymentDetails() },
+            style: style
+        )
+        setUpFields(for: component)
+        component.delegate = self
+        return component
+    }()
+    
+    /// :nodoc:
+    public lazy var viewController: UIViewController = formComponent.viewController
+    
+    /// :nodoc:
+    /// Constructs the fields for the form based on the configuration
+    private func getFormFields() -> [PersonalInformation] {
+        var fields: [PersonalInformation] = [
+            .custom(CustomFormItemInjector(item: headerFormItem(key: .boletoPersonalDetails))),
+            .firstName,
+            .lastName,
+            .custom(CustomFormItemInjector(item: socialSecurityNumberItem))
+        ]
+
+        if configuration.shopperInfo.billingAddress != nil {
+            fields.append(
+                .custom(CustomFormItemInjector(item: headerFormItem(key: .billingAddressSectionTitle)))
+            )
+            fields.append(.custom(CustomFormItemInjector(item: billingAddressLabelItem)))
+        } else {
+            fields.append(.address)
+        }
+        
+        if configuration.showEmailAddress {
+            fields.append( .custom(CustomFormItemInjector(item: sendCopyByEmailItem)))
+            fields.append(.email)
+        }
+        
+        return fields
+    }
+    
+    /// :nodoc:
+    /// Sets the initial values for the form fields based on configuration
+    private func setUpFields(for component: FormComponent) {
+        if let shopperName = configuration.shopperInfo.shopperName {
+            component.firstNameItem?.value = shopperName.firstName
+            component.lastNameItem?.value = shopperName.lastName
+        }
+        
+        if let socialSecurityNumber = configuration.shopperInfo.socialSecurityNumber {
+            socialSecurityNumberItem.value = socialSecurityNumber
+        }
+        
+        if let emailItem = component.emailItem {
+            sendCopyByEmailItem.value = false
+            emailItem.value = configuration.shopperInfo.emailAddress ?? ""
+            bind(sendCopyByEmailItem.publisher, to: emailItem, at: \.isHidden.wrappedValue, with: { !$0 })
+        }
+    }
+    
+    /// :nodoc:
+    private func createPaymentDetails() -> PaymentMethodDetails {
+        guard let firstNameItem = formComponent.firstNameItem,
+              let lastNameItem = formComponent.lastNameItem,
+              case let shopperName = ShopperName(firstName: firstNameItem.value, lastName: lastNameItem.value),
```

Comment:
Why there is `case`? 🤔

Reply:
wanted to create all of the data beforehand. i have moved it later in the function

Reply:
I mean from syntax point of view:

`case` is for enums, right? What is enum in this line ?